src/session/session.c

```diff
@@ -914,8 +915,8 @@ main (int argc,
 
   program_name = basename (argv[0]);
 
-  /* Cleanup the umask */
-  umask (077);
+  /* Tight umask during authentication */
+  mode_t orig_umask = umask (077);
```

Comment:
I checked: The only direct writing open() call that cockpit-session does is this:

    open (_PATH_BTMP, O_WRONLY | O_APPEND);

However:

 - it usually exists already, hence the umask/mode shouldn't matter; when it's deleted, /usr/lib/tmpfiles.d/var.conf restores the permissions on the next boot.
 - the call does *not* specify `O_CREAT`, so if it does not exist, it would just get an ENOENT
 - the file has root:utmp 0660 permissions, so creating it with either 077 or 022 umasks would both be wrong. 

So I think this call is not relevant, after all.

The only other place that could write files that I can think of is if GSSAPI writes a kerberos ticket cache file (yes, it normally goes to the kernel keyring or the daemon, but `$KRB5CCNAME` is still a thing). World-readable files would indeed be disastrous there. But [krb5](https://github.com/krb5/krb5/blob/master/src/lib/krb5/rcache/rc_dfl.c#L103) does the right thing already and open these with a `0600` mask (it has to, otherwise every `kinit` invocation would be vulnerable).

So in summary, I don't see anything which would go wrong with 022 permissions.

Reply:
I think we should just leave the umask alone, then.